File: darkwater-core/src/main/java/kiinse/plugins/darkwaterapi/core/commands/DarkContext.java
package kiinse.plugins.darkwaterapi.core.commands;

import kiinse.plugins.darkwaterapi.api.commands.CommandContext;
import kiinse.plugins.darkwaterapi.api.files.locale.PlayerLocale;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

public class DarkContext implements CommandContext {

    private final CommandSender sender;
    private final PlayerLocale playerLocale;
    private final String[] args;

    public DarkContext(CommandSender sender, PlayerLocale playerLocale, String[] args) {
        this.sender = sender;
        this.playerLocale = playerLocale;
        this.args = args;
    }

    @Override
    public @NotNull CommandSender getSender() {
        return sender;
    }

    @Override
    public @NotNull String[] getArgs() {
        return Arrays.copyOf(args, args.length);
    }

    @Override
    public @NotNull PlayerLocale getSenderLocale() {
        return playerLocale;
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof CommandContext)) return false;
        return obj.hashCode() == hashCode();
    }

    @Override
    public String toString() {
        return "Sender: " + getSender().getName() + "\n" +
               "Locale: " + getSenderLocale() + "\n" +
               "Args: " + Arrays.toString(getArgs());
    }

    @Override
    public int hashCode() {
        return toString().hashCode();
    }
}

File: core/src/main/kotlin/kiinse/me/plugins/darkwaterapi/core/commands/DarkContext.kt
package kiinse.me.plugins.darkwaterapi.core.commands

import kiinse.me.plugins.darkwaterapi.api.commands.CommandContext
import kiinse.me.plugins.darkwaterapi.api.files.locale.PlayerLocale
import org.bukkit.command.CommandSender

class DarkContext(override val sender: CommandSender, override val senderLocale: PlayerLocale, override val args: Array<String>) : CommandContext {

    override fun equals(other: Any?): Boolean {
        return other != null && other is CommandContext && other.hashCode() == hashCode()
    }

    override fun toString(): String {
        return """
            Sender: ${sender.name}
            Locale: $senderLocale
            Args: ${args.contentToString()}
            """.trimIndent()
    }

    override fun hashCode(): Int {
        return toString().hashCode()
    }
}